### PR TITLE
Option to Skip added

### LINKED ISSUE #9: Include a skip option
Include a skip/none option. On selection of this. you don't add anything to config.

### PR DESCRIPTION
**Pull request by awebdeveloper**
Skip option added. This should fix #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,9 @@ Code example.
 #### options.steps[].variants[].push {Boolean}
 Applicable in case of array. Default behaviour, is to replace.  Setting it true will force the value to be pushed instead.
 
+#### options.steps[].variants[].skip {Boolean}
+Skip true will not add any value to array
```
Comment (gwer) [inline README.md]: It can be any value, not only array.

File: oncogene.js [hunk 2]
```diff
@@ -257,8 +261,8 @@ class Oncogene {
                     throw new Error(`Variant ${variantNum} of step ${stepNum} should have hint or code`)
                 }
 
-                if (this.isUndefined(variant.value)) {
-                    throw new Error(`Variant ${variantNum} of step ${stepNum} should have a value`)
+                if (this.isUndefined(variant.value) && this.isUndefined(variant.skip)) {
```
Comment (gwer) [inline oncogene.js]: What if `variant.skip === false`?

**Review by gwer — CHANGES_REQUESTED** [review]

File: oncogene.js (hunk 2 above)
Comment (awebdeveloper) [reply oncogene.js]: this is just validation right ??

File: README.md (hunk 1 above)
Comment (awebdeveloper) [reply README.md]: fixed

File: oncogene.js (hunk 2 above)
Comment (gwer) [reply oncogene.js]: That's right, but if `variant.value` is undefined then `variant.skip` should be true.

Comment (awebdeveloper) [inline oncogene.js]: here we are just validating that one of the two should be present

Comment (awebdeveloper) [reply oncogene.js]: k done

File: oncogene.js [hunk 3]
```diff
@@ -56,6 +56,10 @@ class Oncogene {
 
     setVal(key, selectedVarient) {
         const step = this.getStep()
+        if(selectedVarient.skip === true) {
```
Comment (gwer) [inline oncogene.js]: Let's allow any truthy value:
`if (selectedVarient.skip) return`

File: oncogene.js [hunk 4]
```diff
@@ -257,8 +261,8 @@ class Oncogene {
                     throw new Error(`Variant ${variantNum} of step ${stepNum} should have hint or code`)
                 }
 
-                if (this.isUndefined(variant.value)) {
-                    throw new Error(`Variant ${variantNum} of step ${stepNum} should have a value`)
+                if (this.isUndefined(variant.value) && (this.isUndefined(variant.skip) || variant.skip === false)) {
```
Comment (gwer) [inline oncogene.js]: And here we will check that variant skip is not falsy:
`if (this.isUndefined(variant.value) && !variant.skip)`

**Review by gwer — CHANGES_REQUESTED** [review]

File: oncogene.js (hunk 3 above)
Comment (awebdeveloper) [reply oncogene.js]: I personally prefer explicit checks but if you don't I will change

Explicit values helps in testing and prevents user mistakes

Comment (gwer) [reply oncogene.js]: It depends, but even if you want check strict equality then in validation you should check that `variant.skip !== true`. User can set `variant.skip = 0` for example.

Comment (awebdeveloper) [reply oncogene.js]: fixed

File: oncogene.js [hunk 5]
```diff
@@ -257,8 +261,8 @@ class Oncogene {
                     throw new Error(`Variant ${variantNum} of step ${stepNum} should have hint or code`)
                 }
 
-                if (this.isUndefined(variant.value)) {
-                    throw new Error(`Variant ${variantNum} of step ${stepNum} should have a value`)
+                if (this.isUndefined(variant.value) && (this.isUndefined(variant.skip) || variant.skip)) {
```
Comment (gwer) [inline oncogene.js]: `variant.skip == true` shouldn't cause error.

Try this: `if (this.isUndefined(variant.value) && !variant.skip)`

Comment (awebdeveloper) [reply oncogene.js]: thanks .... done

File: oncogene.js [hunk 6]
```diff
@@ -261,7 +261,7 @@ class Oncogene {
                     throw new Error(`Variant ${variantNum} of step ${stepNum} should have hint or code`)
                 }
 
-                if (this.isUndefined(variant.value) && (this.isUndefined(variant.skip) || variant.skip)) {
+                if (this.isUndefined(variant.value) && (this.isUndefined(variant.skip) || !variant.skip)) {
```
Comment (gwer) [inline oncogene.js]: `!variant.skip` includes a check for undefined.

**Review by gwer — COMMENTED** [review]
Run `npm run build` and I will merge this PR.

File: oncogene.js (hunk 6 above)
Comment (awebdeveloper) [reply oncogene.js]: that's y i was checking for undefined || true. Any case updated the pr

**Pull request merged**